### ISSUE TITLE
Cannot build project with OR-Tools and Mediapipe as external dependencies

### ISSUE THREAD

**Issue by VladYushchenko** [google/or-tools#2634]
**System information**
    OS: Ubuntu 18.04
    Programming Language and version: C++
    MediaPipe version: 0.8.0
    OR-Tools version: v9.0 / v8.0
    Bazel: 4.1.0

**Solver**
 Hungarian algorithm from "ortools/algorithms/hungarian.h"

**Setup**

Hi, I have a problem with integration of both Mediapipe and OR-Tools in the same repository.

I have opened the same issue in https://github.com/google/mediapipe/issues/2258
At first glance, it is hard to tell what the exact cause is, so I believe it makes sense to report it in both repos.

So yes, I use OR-Tools in my SORTTracker implementation, which builds successfully.
I use Mediapipe and SORTTracker as dependencies in another project. 

Mediapipe and OR-Tools are included in Bazel's BUILD file as following:

```
git_repository(
    name = "com_google_ortools",
    commit = "8d19323", # release v8.0
    remote = "https://github.com/google/or-tools.git",
)

_MEDIAPIPE_VERSION = "0.8.0"

http_archive(
    name = "mediapipe",
    sha256 = "0b129a28864fd2cd2822948621645efb9d62b6d043d1c4da3cf2d01886bfd901",
    strip_prefix = "mediapipe-{}".format(_MEDIAPIPE_VERSION),
    url = "https://github.com/google/mediapipe/archive/{}.tar.gz".format(_MEDIAPIPE_VERSION),
)
```

**Problem**

If I use Mediapipe 0.8.0 with the recent OR-Tools v9.0, but the build fails with a Glog error 
[v9.0_GlogError.txt](https://github.com/google/or-tools/files/6777987/v9.0_GlogError.txt)

As suggested in https://github.com/google/or-tools/issues/2378, the latter can be only resolved by downgrading OR-Tools to v8.0

Consequently, switching to v8.0 induces various typedef errors between OR-Tools and Mediapipe 
[v8.0_TypeError.txt](https://github.com/google/or-tools/files/6777990/v8.0_TypeError.txt)

**Expected behavior**
I would expect that at least one configuration would work, but for me it seems to be a deadlock.

Thanks!

**Comment by lperron** [comment]
Yes. We have reimplemented glog inside or-tools on top of abseil. This will be resolved when abseil reimplements the logging facility.

**Comment by lperron** [comment]
Duplicate of #2378

**Comment by VladYush** [comment]
Thanks for a quick answer @lperron 

What about the typedef error when using or-tools v8.0? 
Is it also caused by reimplementation of some parts and cannot be fixed in the near future?

Cheers!

**Comment by VladYushchenko** [comment]
Thanks for a quick answer @lperron 

What about the typedef error when using or-tools v8.0? 
Is it also caused by reimplementation of some parts and cannot be fixed in the near future?

Cheers!

**Comment by Mizux** [comment]
For v9 issue:
As Laurent said google/glog currently depends on GFlags which is deprecated and unsupported by Google (only maintained by the community AFAIK). Abseil-cpp "just" released (few month ago?) the flag component so we lazy patch glog source to use absl:flag instead of Gflags -> copy paste src -> hack them -> keep the same namespace -> conflict with project still pulling glog :disappointed: 
1. We don't know when an [g]log component will make it through abseil-cpp (already waiting more than a year IIRC) if it make it.
2. Protobuf also provide it own logging component with the advantage to provide a logHandler (https://github.com/protocolbuffers/protobuf/blob/3f5fc4df1de8e12b2235c3006593e22d6993c3f5/src/google/protobuf/stubs/logging.h#L203-L219)

A temporary solution would be to scope our glog implementation in `namespace operations_research` and adapt the code everywhere BUT we currently don't have any human resource to do it...

For v8 issue:
basically both project use the same constants value which are located in the global namespace in both side.
-> so you need to fix one the project (by creating a patch ?) so they can use the same header.

FYI Google is moving to C++17 and <cstdint>/<limits> so we can expect in the future that both project won't use these typedef and will only rely on C++/STL aka `std::numeric_limits<int16_t>::max()`.
FYI we already start this cleanup effort on our side (OR-Tools)
